model/ecoflow/powerstream.py

```diff
@@ -65,8 +65,8 @@ def customize_homie(self):
         _LOGGER.debug(f"property {self.today_from_solar.name} has been added to node {node.name}")
 
     def handle_energy_total_report(self, pdata, header):
-        # [total, ?, ?, from_bat, pv1?, pv2?]
-        sums = [0,0,0,0,0,0]
+        # [total delivery?, to plugs?, to battery?, from_bat, ?,?, pv1?, pv2?] it is in total 8 values in the report watth1...watth8
+        sums = [0,0,0,0,0,0,0,0]
```

Comment:
This order does not seem to fit, currently I do not have a battery attached to my powerstream to the only source of power are pv1+2 and the last two values are 0 here. So it looks more like:
`[total, to plugs?, to battery?, from_bat, pv1, pv2, ?,?]`

But I will do more testing during the day, to get more values and a better overview.

Reply:
May be the code needs some enhancements to cover my observation, the proto of the EnergyPack has repeated but optional values and looking at the transmitted and decoded message it is indeed 6 values. Have a look to the raw message, the numbering the 5th and 6th is omitted and solar1/2 are the last two. 

I did not have a full telegram so far, but my setup may not use all the options.
So my interpretation is so far:
watthType=1= to house loads
watthType=2= to plugs
watthType=3= to battery
watthType=4= from battery,
watthType=5= ?
watthType=6= ?
watthType=7= from solar PV1
watthType=8= from solar PV2
Most likely my code adjustment needs a more sophisticated mapping to cover the empty (not send) values.
But for now it may be OK to keep the 6 items and waiting for the moment that all 8 are transmitted

Reply:
I think we are interpreting the energy-protobuf message differently? I use this proto definition:
```proto

message EnergyItem
{
    optional uint32 timestamp = 1;
    optional uint32 watth_type = 2;
    repeated uint32 watth = 3;
}

message EnergyTotalReport
{
    optional uint32 watth_seq = 1;
    optional EnergyItem watth_item = 2;
}
```

which results in a decoded message like this:
```json
"watthSeq": 65535,
        "watthItem": {
            "timestamp": 1700265600,
            "watthType": 8,
            "watth": [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                35,
                91,
                19,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                18,
                49,
                10,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                17,
                46,
                10,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0
            ]
        }
        ```
        
       as you can see there are 144 watth-items, which I interpreted as 6 different values for every hour of today.
       The code sums these values and the results seem to match the values I see in the ecoflow app.

Reply:
yes, I am using

message EnergyPack
{
    optional uint32 sys_seq = 1;
    repeated EnergyItem sys_energy_stream = 2;
}
but in the end it is the same, because it references back to the repeated EnergyItem

the 6th transmitted EnergyItem is called:
"watthType": 8,
and "watthType": 5 and "watthType": 6 are not part of the transmitted telegram. But may exist at a certain time.

I will check my transmitted values per watthType, how many these are.
I also sum up the values to get the number presented in the app.
But I have for each watthType own row of values.

Reply:
I have 24 values per type

```
{"sysSeq":1,"sysEnergyStream":[{"timestamp":1693785600,"watthType":1,"watth":[0,0,0,0,0,11,0,0,0,0,0,0,0,0,0,0,0,0,101,101,101,101,101,101]},{"timestamp":1693785600,"watthType":2,"watth":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]},{"timestamp":1693785600,"watthType":3,"watth":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]},{"timestamp":1693785600,"watthType":4,"watth":[0,0,0,0,0,11,0,0,0,0,0,0,0,0,0,0,0,0,100,100,100,100,101,101]},{"timestamp":1693785600,"watthType":7,"watth":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]},{"timestamp":1693785600,"watthType":8,"watth":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]}]}
```

Reply:
Yes I will also use your way, seems to be the right one.